pyomo/contrib/preprocessing/plugins/equality_propagate.py

```diff
@@ -0,0 +1,226 @@
+            if (len(vars_) == 2 and
+                    sorted(l for l in repn.linear if l) == [-1, 1]):
+                # this is an a == b constraint.
+                v1 = vars_[0]
+                v2 = vars_[1]
+                set1 = eq_var_map.get(v1, ComponentSet([v1]))
+                set2 = eq_var_map.get(v2, ComponentSet([v2]))
+                relevant_vars.update([v1, v2])
+                set1.update(set2)  # set1 is now the union
+                for v in set1:
+                    eq_var_map[v] = set1
+
+    return eq_var_map, relevant_vars
+
+
+def _detect_fixed_variables(m):
+    """Detect fixed variables due to constraints of form var = const."""
+    new_fixed_vars = ComponentSet()
+    for constr in m.component_data_objects(ctype=Constraint,
+                                           active=True,
+                                           descend_into=True):
+        if constr.equality and constr.body.polynomial_degree() == 1:
+            repn = generate_canonical_repn(constr.body)
+            if len(repn.variables) == 1 and repn.linear[0]:
+                var = repn.variables[0]
+                coef = float(repn.linear[0])
+                const = repn.constant if repn.constant is not None else 0
+                var_val = (value(constr.lower) - value(const)) / coef
+                var.fix(var_val)
+                new_fixed_vars.add(var)
+    return new_fixed_vars
+
+
+class FixedVarPropagator(IsomorphicTransformation):
+    """Propagates variable fixing for equalities of type x = y.
+
+    If x is fixed and y is not fixed, then this transformation will fix y to
+    the value of x.
+
+    This transformation can also be performed as a temporary transformation,
+    whereby the transformed variables are saved and can be later unfixed.
+
+    """
+
+    alias('contrib.propagate_fixed_vars',
+          doc=textwrap.fill(textwrap.dedent(__doc__.strip())))
+
+    def __init__(self):
+        """Initialize the transformation."""
+        super(FixedVarPropagator, self).__init__()
+
+    def _apply_to(self, instance, tmp=False):
+        """Apply the transformation.
+
+        Args:
+            instance (Block): the block on which to search for x == y
+                constraints. Note that variables may be located anywhere in
+                the model.
+            tmp (bool, optional): Whether the variable fixing will be temporary
+
+        Returns:
+            None
+
+        Raises:
+            ValueError: if two fixed variables x = y have different values.
+
+        """
+        if tmp and not hasattr(instance, '_tmp_propagate_fixed'):
+            instance._tmp_propagate_fixed = ComponentSet()
+        eq_var_map, relevant_vars = _build_equality_set(instance)
+        #: ComponentSet: The set of all fixed variables
+        fixed_vars = ComponentSet((v for v in relevant_vars if v.fixed))
+        newly_fixed = _detect_fixed_variables(instance)
```

Comment:
I think you should remove this line from this transformation since you have a separate transformation for detecting fixed variables and right now it seems like an undocumented side-effect for this transformation. I think the user should have to manually apply these transformations in a sequence.

Reply:
This transformation would not do anything without the other one. The moment that you have `x=y` and fix `x=4`, the original constraint becomes `y=4`, and you aren't able to detect the `x=y` equality link.